src/api/app/models/event/base.rb

```diff
@@ -105,9 +107,10 @@ def receiver_roles
     end
 
     def initialize(attribs)
-      attributes = attribs.dup
+      attributes = attribs.dup.with_indifferent_access
       super()
       self.created_at = attribs[:time] if attributes[:time]
+      self.package = Package.get_by_project_and_name(attributes[:project], attributes[:package]) if attributes[:project] && attributes[:package]
```

Comment:
This will make the whole Event creation barf if this goes wrong (there are several exceptions raised inside this method). Also many of the events (branch_command or build*) could be about packages that come over project links and/or from remote so there is no object to get.